test/avocado/testlib_avocado/machineslib.py

```diff
@@ -166,3 +167,87 @@ def tearDown(self):
         super().tearDown()
         if self.virshvm:
             self.destroy_vm(self.virshvm)
+        if len(self.vmlist) != 0:
+            for vm in self.vmlist:
+                self.destroy_vm(vm)
+        if self.machine.execute("sudo virsh pool-list | grep apool", raising=False) != '' and self.machine.execute("virsh pool-list | grep mypool", raising=False) != '':
+            self.machine.execute(
+                'sudo virsh vol-delete apooldisk1 apool \
```

Comment:
If you don't want to use vol-delete for some specific reason, you can just destroy the pool (aka deactivate it) and them do rm -rf the pool directory, omiting all vol-delete commands.